File: src/main/java/com/bombproj/controller/ProjectFilesController.java
package com.bombproj.controller;

import com.bombproj.dto.ProjectFileDto;
import com.bombproj.framework.JsonResult;
import com.bombproj.framework.SessionConfig;
import com.bombproj.framework.interceptor.AuthPassport;
import com.bombproj.service.ProjectFilesService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController("/api/project")
public class ProjectFilesController {

    @Autowired
    private ProjectFilesService projectFilesService;

    @AuthPassport
    @GetMapping("{projectId}/folders/{folderId}")
    public JsonResult listFiles(@PathVariable("projectId") String projectId,
                                @PathVariable("folderId") String folderId,
                                @ModelAttribute ProjectFileDto dto) {
        dto.setProjectId(projectId);
        dto.setFolderId(folderId);
        return JsonResult.success(this.projectFilesService.pageProjectFiles(dto));
    }

    @AuthPassport
    @PostMapping("{projectId}/folders/{folderId}/mkdir")
    public JsonResult mkdir(@PathVariable("projectId") String projectId,
                            @PathVariable("folderId") String folderId,
                            @ModelAttribute ProjectFileDto dto) {
        SessionConfig sessionConfig = SessionConfig.current();
        dto.setProjectId(projectId);
        dto.setFolderId(folderId);
        dto.setCreatUserId(sessionConfig.getUserId());
        return JsonResult.success();
    }

}

File: src/main/scala/com/bombproj/controller/ProjectFilesController.scala
package com.bombproj.controller

import com.bombproj.dto.ProjectFileDto
import com.bombproj.framework.interceptor.AuthPassport
import com.bombproj.framework.{JsonResult, SessionConfig}
import com.bombproj.service.ProjectFilesService
import javax.annotation.Resource
import org.springframework.web.bind.annotation.{GetMapping, ModelAttribute, PathVariable, PostMapping, RequestMapping, RestController}

@RestController
@RequestMapping(Array("/api/project"))
class ProjectFilesController @Resource()(projectFilesService: ProjectFilesService) {

    @AuthPassport
    @GetMapping(Array("{projectId}/folders/{folderId}"))
    def listFiles(@PathVariable("projectId") projectId: String,
                  @PathVariable("folderId") folderId: String,
                  @ModelAttribute dto: ProjectFileDto): JsonResult = {
        dto.setProjectId(projectId)
        dto.setFolderId(folderId)
        JsonResult.success(data = this.projectFilesService.pageProjectFiles(dto))
    }

    @AuthPassport
    @PostMapping(Array("{projectId}/folders/{folderId}/mkdir"))
    def mkdir(@PathVariable("projectId") projectId: String,
              @PathVariable("folderId") folderId: String,
              @ModelAttribute dto: ProjectFileDto): JsonResult = {
        val sessionConfig = SessionConfig.current
        dto.setProjectId(projectId)
        dto.setFolderId(folderId)
        dto.setCreatUserId(sessionConfig.userId)
        JsonResult.success()
    }

}
